client/src/App.vue

```diff
@@ -0,0 +1,177 @@
+<template>
+<div id="app">
+    <div class="panel panel-default">
+      <div class="panel-heading"></div>
+      <div class="row">
+        <div class="search-wrapper panel-heading col-sm-12">
+          <input class="form-control" type="text" v-model="searchQuery" placeholder="Search" />
+        </div>
+      </div>
+      <div class="table-responsive">
+        <table v-if="stores.length" class="table" bordered>
+          <thead>
+            <tr>
+              <th>Id</th>
+              <th>Code</th>
+              <th>Description</th>
+              <th>Name</th>
+              <th>OpeningDate</th>
+              <th>StoreType</th>
+              <th>Season</th>
+              <th>Action</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr v-for="store in resultQuery" :key="store.name">
+              <td>{{store.id}}</td>
+              <td>{{store.code}}</td>
+              <td>{{store.description}}</td>
+              <td contenteditable="true"><input v-model="store.name"></td>
+              <td>{{store.openingDate}}</td>
+              <td>{{store.storeType}}</td>
+              <td>{{store.season}}</td>
+              <td>
+              <span>
+                <button type="button" class="btn btn-danger btn-rounded btn-sm my-0"  v-on:click="saveStore(store)">
+                  Save
+                </button></span>
+            </td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    </div>
+    <button type="button" class="btn btn-info action_btn" v-on:click="downloadCSVData">
+      Download
+</button>
+  </div>
+</template>
+
+<script>
+
+
+import axios from 'axios'
+
+export default {
+  name: "App",
+  data() {
+    return {
+        searchQuery: null,
+        stores:[]
+    };
+  },
+  beforeCreate(){
+
+    var self = this
+    const storesUrl = 'http://localhost:8090/stores'
+    const saveStoreDataInDatabaseUrl = "http://localhost:8090/saveStoreDataInDatabase"
+    const saveStoreAndSeasonDataInDatabaseurl = "http://localhost:8090/saveStoreAndSeasonDataInDatabase"
+
+      axios.get(saveStoreDataInDatabaseUrl, {
```

Comment:
Every time the frontend is requested, it will start this routine that fetches all the stores from the API.

What are the pros and cons of this design decision, and what difference would it make to use the Spring Scheduler instead of this approach.

Reply:
Pros :
  That data latest data is available on frontend.
Cons : 
-Every time, we refresh the page, it will again hit the api, even in the case, where the previous call might not have completed. So that might cause performance issues.
- If api changes, need to modify them on frontend. 

Yes, We can use spring scheduler to schedule the call on equal intervals to provided endpoints, to sync data with database. It will be independent to frontend. Frontend will simply fetch , what is present in database. Need not to take care to sync endpoint data with database.

Reply:
What happens if 10 users access the app simultaneously?

Reply:
Spring handles it automatically by creating seperate thread for each user.

Reply:
Indeed it does, but that does not fix the problem of 10 concurrent fetches and database writes. Can you see how fast this would reach a situation where the app would just enter into a DoS state, specially if you were fetching all the expected pages from the challenge API?

Reply:
Yes, This will create a problem. I could have implemented it in better way. This should have been run as standalone singleton service to tackle this.